spring/boot-autoconfigure/src/main/java/com/linecorp/armeria/spring/GrpcServiceRegistrationBean.java

```diff
@@ -92,4 +99,21 @@ public GrpcServiceRegistrationBean addExampleRequest(@NotNull String serviceType
                                                          @NotNull Object exampleRequest) {
         return addExampleRequest(GrpcExampleRequest.of(serviceType, methodName, exampleRequest));
     }
+
+    /**
+     * Returns example {@link GrpcExampleHeaders}.
+     */
+    @NotNull
+    public Collection<GrpcExampleHeaders> getExampleHeaders() {
+        return exampleHeaders;
+    }
+
+    /**
+     * Sets example {@link GrpcExampleHeaders}.
+     */
+    public GrpcServiceRegistrationBean setExampleHeaders(
+            @NotNull Collection<GrpcExampleHeaders> exampleHeaders) {
+        this.exampleHeaders.addAll(exampleHeaders);
```

Comment:
I find it is somewhat confusing to add in a setter method. How about adding `addExampleHeaders(GrpcExampleHeaders)` and `addExampleHeaders(Iterable<GrpcExampleHeaders>)` to add and doing `this.exampleHeaders = ImmutableList.copyOf(exampleHeaders);` here instead?

This could be applied to all other registration beans and all their counterparts that handle the example requests, i.e.

- Add `addExampleRequests(*ExampleRequests)` and `addExampleRequests(Iterable<*ExampleRequests>)` to all registration beans.
- Add `addExampleHeaders(*ExampleHeaders)` and `addExampleHeaders(Iterable<*ExampleHeaders>)` to all registration beans.
- Make `setExample{Requests,Headers}()` set, not add.

Reply:
Shouldn't we consider the following situations?

```java
public GrpcServiceRegistrationBean setExampleHeaders(Iterable<GrpcExampleHeaders> exampleHeaders) {
    this.exampleHeaders = ImmutableList.copyOf(exampleHeaders);
}

// ...

new GrpcServiceRegistrationBean()
...
.setExampleHeaders(...)
.addExampleHeaders(...);
```

In this case, the user will see `UnsupportedOperationException()`.

Reply:
That was a bad idea that adding example requests to the existing requests when calling `setExampleRequests`. Sorry about that. 🤦‍♂ 

> In this case, the user will see UnsupportedOperationException().

To avoid this, we should use a sort of `builder` for that if there're existing headers added:
```java
this.exampleHeaders = ImmutableList.<HttpHeaders>builder().addAll(this.exampleHeaders)
                                                          .addAll(exampleHeaders).build();
```

Reply:
You're welcome! And thank you for the good idea. 👍 
I will resend the PR.